docs/proposals/26-workload-identity.md

```diff
@@ -0,0 +1,547 @@
+
+```yaml
+apiVersion: authentication.gardener.cloud/v1alpha1
+kind: TokenRequest
+spec:
+  contextObject: # Optional field, various metadata about context of use of the token
+    apiVersion: core.gardener.cloud/v1beta1
+    kind: Shoot
+    name: foo
+    namespace: garden-local
+    uid: 54d09554-6a68-4f46-a23a-e3592385d820
+status:
+  token: eyJhbGciOiJ....OkBBrVWA # The generated OIDC token
+  expirationTimestamp: 2024-02-09T16:35:02Z
+```
+
+#### Shoot API Related Changes
+
+Currently, shoot clusters set the infrastructure credentials via an intermediate
+resource named `SecretBinding` which is referring to the actual Kubernetes
+secret that contains the static credentials. If `SecretBinding` is extended to
+refer `WorkloadIdentity` as infrastructure credentials, from user experience
+point of view `SecretBinding` is not the best name for such resource, because it
+is no longer limited to referring only secrets as its name implies. Therefore, a
+new resource named `CredentialsBinding` in the API group
+`authentication.gardener.cloud` will be implemented. It will have all features
+of `SecretBinding`, but on top of that will be extended to refer to
+`WorkloadIdentity` resources via `.workloadIdentityRef` field.
+`CredentialsBinding` will be allowed to set exactly one of the `.secretRef` or
+`.workloadIdentityRef` fields, but not both or none of them.
+
+In a nutshell, the changes introduced compared to `SecretBinding` are:
+
+- `CredentialsBinding.secretRef` field will be optional and mutable.
+- `CredentialsBinding.workloadIdentityRef` field will be optional and mutable.
+  It will refer to a `WorkloadIdentity` resource by its name and namespace. If
+  the namespace is unset, the namespace of the `CredentialsBinding` is
+  will be used.
+- `quotas` and `provider` fields have the semantic as their respective
+  counterparts in the `SecretBinding` API. `providers` will be made mandatory
+  field also via the API specification.
+- Static validation will ensure that exactly one of `secretRef` or
+  `workloadIdentityRef` fields is set, but not both or none of them.
+- On update of the `workloadIdentityRef`, extension admission controller should
+  ensure that both the old and the new `WorkloadIdentity` are for the same cloud
+  provider account, if such validation is possible for the given extension.
+
+```yaml
+apiVersion: authentication.gardener.cloud/v1alpha1
+kind: CredentialsBinding
+metadata:
+  name: my-credentials
+  namespace: garden-local
+provider:
+  type: aws # {aws,azure,gcp,...}
+secretRef: # unlike SecretBindings, this field will be optional and mutable
+  name: static
+  # namespace: "...", allow reference across namespaces
+workloadIdentityRef:
+  name: banana-testing
+  # namespace: "...", allow reference across namespaces
+quotas: []
+# - name: quota-1
+# # namespace: garden-quoatas
+```
+
+Shoot API will be extended with new field `.spec.credentialsBindingName` which
```

Comment:
```suggestion
Shoot API will be extended with new field `.spec.credentialsBindingName` whose
```

Reply:
I am not a native speaker but I have the gut feeling `whose` is used with persons, and `which` with things, but I will apply the proposed change.